Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -13,50 +13,40 @@ import Foundation
 
 open class ChartData: NSObject, ExpressibleByArrayLiteral
 {
-    internal var _yMax: Double = -Double.greatestFiniteMagnitude
-    internal var _yMin: Double = Double.greatestFiniteMagnitude
-    internal var _xMax: Double = -Double.greatestFiniteMagnitude
-    internal var _xMin: Double = Double.greatestFiniteMagnitude
-    internal var _leftAxisMax: Double = -Double.greatestFiniteMagnitude
-    internal var _leftAxisMin: Double = Double.greatestFiniteMagnitude
-    internal var _rightAxisMax: Double = -Double.greatestFiniteMagnitude
-    internal var _rightAxisMin: Double = Double.greatestFiniteMagnitude
+
+    @objc public internal(set) var xMax = -Double.greatestFiniteMagnitude
+    @objc public internal(set) var xMin = Double.greatestFiniteMagnitude
+    @objc public internal(set) var yMax = -Double.greatestFiniteMagnitude
+    @objc public internal(set) var yMin = Double.greatestFiniteMagnitude
+    var leftAxisMax = -Double.greatestFiniteMagnitude
+    var leftAxisMin = Double.greatestFiniteMagnitude
+    var rightAxisMax = -Double.greatestFiniteMagnitude
+    var rightAxisMin = Double.greatestFiniteMagnitude
     
-    internal var _dataSets = [ChartDataSetProtocol]()
+    var _dataSets = [Element]()
     
     public override required init()
     {
         super.init()
     }
 
-    public required init(arrayLiteral elements: ChartDataSetProtocol...)
+    public required init(arrayLiteral elements: Element...)
     {
         super.init()
-
-        _dataSets = elements
-
-        self.initialize(dataSets: _dataSets)
+        self.dataSets = elements
     }
 
-    @objc public init(dataSets: [ChartDataSetProtocol]?)
+    @objc public init(dataSets: [Element])
     {
         super.init()
-        
-        _dataSets = dataSets ?? [ChartDataSetProtocol]()
-        
-        self.initialize(dataSets: _dataSets)
+        self.dataSets = dataSets
     }
     
-    @objc public convenience init(dataSet: ChartDataSetProtocol?)
+    @objc public convenience init(dataSet: Element)
     {
-        self.init(dataSets: dataSet === nil ? nil : [dataSet!])
-    }
-    
-    internal func initialize(dataSets: [ChartDataSetProtocol])
```

Comment:
I didn't see a replacement for this func? And I didn't see there is a function calling notifyDataChanged() somewhere else.

Reply:
This function would cause `notifyDataChanged` to be called twice. Once here and once in the setter of `dataSets`